src/wallet/walletdb.cpp

```diff
@@ -997,6 +1001,9 @@ static DBErrors LoadDescriptorWalletRecords(CWallet* pwallet, DatabaseBatch& bat
         // Only log if there are no critical errors
         pwallet->WalletLogPrintf("Descriptors: %u, Descriptor Keys: %u plaintext, %u encrypted, %u total.\n",
                desc_res.m_records, num_keys, num_ckeys, num_keys + num_ckeys);
+
+        // Upgrade the wallet to have a global HD key
```

Comment:
I prefer these checks to be in the function itself rather than the caller for these upgrades as we may want to call these from elsewhere in the future.